File: src/frontend-new/src/app/todo.ts
export class Todo {
  id!: string;
  title!: string;
  completed!: boolean;
  createdAt!: Date;

  static createNew(): Todo {
    const newTodo = new Todo();
    newTodo.title = '';
    newTodo.completed = false;
    newTodo.createdAt = new Date();
    return newTodo;
  }

  constructor(id?: string, title?: string, completed?: boolean, createdAt?: Date) {
    if (id) this.id = id;
    if (title) this.title = title;
    if (completed !== undefined) this.completed = completed;
    if (createdAt) this.createdAt = createdAt;
    else this.createdAt = new Date();
  }
}

File: src/main/java/com/sukruokul/todo/model/Todo.java
package com.sukruokul.todo.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * author: sukru.okul
 */
@Entity
@Table(name = "todo")
@Document(collection = "todos")
@JsonIgnoreProperties(value = {"createdAt"}, allowGetters = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Todo {

    @Id
    private String id;

    private String title;
    private boolean completed;
    private LocalDateTime createdAt;

    @PrePersist
    public void generateId() {
        if (this.id == null) {
            this.id = UUID.randomUUID().toString();
        }
        if (this.createdAt == null) {
            this.createdAt = LocalDateTime.now();
        }
    }
}